Surface_mesh_simplification/include/CGAL/Surface_mesh_simplification/internal/Common.h

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2006  GeometryFactory (France). All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Fernando Cacciola <fernando.cacciola@geometryfactory.com>
+//
+#ifndef CGAL_SURFACE_MESH_SIMPLIFICATION_DETAIL_COMMON_H
+#define CGAL_SURFACE_MESH_SIMPLIFICATION_DETAIL_COMMON_H
+
+#include <CGAL/license/Surface_mesh_simplification.h>
+
+#include <CGAL/algorithm.h>
+#include <CGAL/boost/graph/properties.h>
+#include <CGAL/boost/graph/iterator.h>
+
+#include <boost/config.hpp>
+#include <boost/iterator_adaptors.hpp>
+#include <boost/optional/optional.hpp>
+#include <boost/tuple/tuple.hpp>
+#include <boost/format.hpp>
+#include <boost/graph/graph_traits.hpp>
+#include <boost/graph/properties.hpp>
+#include <boost/graph/adjacency_list.hpp>
+
+#include <functional>
+#include <utility>
+#include <vector>
+#include <vector>
+#include <set>
+
+namespace CGAL {
+namespace Surface_mesh_simplification {
+
+template<class Handle>
+inline bool handle_assigned(Handle h) { Handle null; return h != null; }
```

Comment:
should be removed as invalid (`null_halfedge()`) should be used.

Reply:
These are not all halfedges but also vertices, edges, etc. which is why it was kept. Could just be removed if `handle_exists` is removed.